data-store/src/main/kotlin/org/ostelco/prime/store/datastore/Schema.kt

```diff
@@ -0,0 +1,100 @@
+package org.ostelco.prime.store.datastore
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.google.cloud.NoCredentials
+import com.google.cloud.Timestamp
+import com.google.cloud.datastore.Blob
+import com.google.cloud.datastore.Datastore
+import com.google.cloud.datastore.DatastoreOptions
+import com.google.cloud.datastore.FullEntity
+import com.google.cloud.datastore.Key
+import com.google.cloud.datastore.KeyFactory
+import com.google.cloud.datastore.LatLng
+import com.google.cloud.datastore.testing.LocalDatastoreHelper
+import com.google.cloud.http.HttpTransportOptions
+import org.ostelco.prime.jsonmapper.objectMapper
+
+
+class EntityStore<T>(
+        private val entityClass: Class<T>,
+        type: String = "inmemory-emulator",
+        namespace: String = "") {
+
+    private val keyFactory: KeyFactory
+    private val datastore: Datastore
+
+    init {
+        datastore = when (type) {
+            "inmemory-emulator" -> {
+                val localDatastoreHelper = LocalDatastoreHelper.create(1.0)
+                localDatastoreHelper.start()
+                localDatastoreHelper.options
+            }
+            "emulator" -> {
+                // When prime running in GCP by hosted CI/CD, Datastore client library assumes it is running in
+                // production and ignore our instruction to connect to the datastore emulator. So, we are explicitly
+                // connecting to emulator
+                // logger.info("Connecting to datastore emulator")
+                DatastoreOptions
+                        .newBuilder()
+                        .setHost("localhost:9090")
+                        .setCredentials(NoCredentials.getInstance())
+                        .setTransportOptions(HttpTransportOptions.newBuilder().build())
+                        .build()
+            }
+            else -> {
+                // logger.info("Created default instance of datastore client")
+                DatastoreOptions
+                        .newBuilder()
+                        .setNamespace(namespace)
+                        .build()
+            }
+        }.service
+        keyFactory = datastore.newKeyFactory().setKind(entityClass.name)
+    }
+
+    fun fetch(key: Key): T {
+        val fullEntity = datastore.fetch(key).single()
+        val map = fullEntity
+                .names
+                .map { name ->
+                    val value = when (entityClass.getDeclaredField(name).type) {
```

Comment:
We will also need some annotation or similar that can be used to mark that a specific value should be "excluded from being indexed".

F.ex. for the String type if indexed then the max size of the string is 1500. If not indexed then the size can be longer. To mark that a string should not be indexed, the value must be constructed using a builder. Example:

```
StringValue.newBuilder(text)
          .setExcludeFromIndexes(true)
          .build()
```
Similar for other datastore supported values.

Reply:
Suggest that we merge this changeset as it is, and then we'll add additional functionality as needed?